### PR TITLE
add openshift/configuration-anomaly-detection

### PR DESCRIPTION
**Pull request by shibumi**
This commit adds code coverage tests for the
openshift/configuration-anomaly-detection repository

SOP for reference: https://github.com/openshift/ops-sop/blob/master/services/codecov.md

### CODE REVIEW

File: ci-operator/config/openshift/configuration-anomaly-detection/configuration-anomaly-detection-master.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+- as: publish-coverage
+  commands: |
+    export CODECOV_TOKEN=$(cat /tmp/secret/CODECOV_TOKEN)
+    make coverage
+  container:
+    from: src
+  postsubmit: true
+  secret:
+    mount_path: /tmp/secret
+    name: configuration-anomaly-detection-codecov-token
```
Comment (georgettica) [inline ci-operator/config/openshift/configuration-anomaly-detection/configuration-anomaly-detection-master.yaml:27]: do we need two actions?

File: ci-operator/config/openshift/configuration-anomaly-detection/configuration-anomaly-detection-master.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  container:
+    from: src
+- as: test
+  commands: make test
+  container:
+    from: src
+- as: validate
+  commands: make validate
+  container:
+    from: src
```
Comment (georgettica) [inline ci-operator/config/openshift/configuration-anomaly-detection/configuration-anomaly-detection-master.yaml:39]: I added the `make ci-check` to hold the right checks for this..
do you think you could change the `ci-check` to do these things?

**Comment by georgettica** [comment]
and to fix the errors you are getting you can `make jobs` or something on these lines in the repo

File: ci-operator/config/openshift/configuration-anomaly-detection/configuration-anomaly-detection-master.yaml (hunk 1 above)
Comment (shibumi) [reply ci-operator/config/openshift/configuration-anomaly-detection/configuration-anomaly-detection-master.yaml:27]: According to the SOP we need both actions. The first is for calculating the coverage the second is for publishing it (I Assume). I might be wrong. I don't understand the CI pipeline configuration TBH.

File: ci-operator/config/openshift/configuration-anomaly-detection/configuration-anomaly-detection-master.yaml (hunk 2 above)
Comment (shibumi) [reply ci-operator/config/openshift/configuration-anomaly-detection/configuration-anomaly-detection-master.yaml:39]: I had to modify our Makefile (https://github.com/openshift/configuration-anomaly-detection/pull/27/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52)

It **must** have the `coverage` PHONY according to the SOP... I guess this is related to Line 21 of this PR. We can remove the other sub tests.. or we put golangci lint in this as well ( right now golangci would be triggered via Github Actions).

File: ci-operator/config/openshift/configuration-anomaly-detection/configuration-anomaly-detection-master.yaml (hunk 1 above)
Comment (georgettica) [reply ci-operator/config/openshift/configuration-anomaly-detection/configuration-anomaly-detection-master.yaml:27]: it's ok, I guess I would add the link to the SOP with this hard requirement.
later I will ask the release team if this is really required

Comment (shibumi) [reply ci-operator/config/openshift/configuration-anomaly-detection/configuration-anomaly-detection-master.yaml:27]: TBH, I think the whole process is a little bit over the top and is from a time where GitHub had no native CI system.
We could just use GitHub Actions... the only thing that's missing is an approval from the Openshift Org admin team.

File: ci-operator/config/openshift/configuration-anomaly-detection/configuration-anomaly-detection-master.yaml (hunk 2 above)
Comment (georgettica) [reply ci-operator/config/openshift/configuration-anomaly-detection/configuration-anomaly-detection-master.yaml:39]: yeah, I think adding `coverage` into `make ci-check` is good and I would want this to be the only entrypoint for CI

this way we have `make` for common dev and `make ci-check` for ci check

we might want to to add `make containerized-ci-check` later on if we want to check why it failed on the CI server

**Comment by shibumi** [comment]
@georgettica  we still have a error :/ I think we are missing something.. maybe a more advanced setup configuration for the cad pipeline in the ci-cluster.

**Comment by georgettica** [comment]
I have another commit I can send you, but maybe you can open your repo so I can push to it:
```
$ git push
ERROR: Permission to shibumi/release.git denied to georgettica.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by shibumi** [comment]
@georgettica  You have access to my branch now.

**Comment by shibumi** [comment]
@smg247  @hongkailiu Is this PR fine? Can we merge this one?

**Comment by georgettica** [comment]
running this automatically assumes you are vendored, so some things fail.
we should specify this repo is not vendored (and maybe have a config toggle so it's clear which repos need it and which don't

**Comment by georgettica** [comment]
`INFO[2022-03-10T09:32:43Z] make: *** No rule to make target `coverage'.  Stop.`

this PR will check against the latest repo and verify the actions works correctly
so you need to merge https://github.com/openshift/configuration-anomaly-detection/pull/27 before this can be merged (or do it without it)

**Comment by shibumi** [comment]
@georgettica  looks like we are still having a few failed tests. Did the pipeline 100% succeed before or did I accidentally implemented a regression while doing the rebase+squash?

**Comment by georgettica** [comment]
it didn't pass 100% (from when I pushed my commit) but the changes I suggested are in my previous comments here

**Comment by shibumi** [comment]
I don't know what's the issue with the failing tests then.. :/

**Comment by hongkailiu** [comment]
> I don't know what's the issue with the failing tests then.. :/

https://coreos.slack.com/archives/CFUGK0K9R/p1646905494473109

**Pull request merged**